src/Microsoft.AspNet.OData.Shared/ETagMessageHandler.cs

```diff
@@ -83,7 +83,7 @@ private static IEdmEntityTypeReference GetTypeReference(IEdmModel model, IEdmEnt
                 return edmTypeReference.AsEntity();
             }
 
-            IEdmTypeReference reference = EdmLibHelpers.GetEdmTypeReference(model, value.GetType());
+            IEdmTypeReference reference = model.GetTypeMappingCache().GetEdmType(value.GetType(), model);
```

Comment:
>model.GetTypeMappingCache(). [](start = 42, length = 28)

Does is it make sense to use GetTypeMappingCache everytime when we try to get the EdmType? If the answer is yes, should we move getting the TypeMapping cache to GetEdmType/GetEdmTypeReference. If no, when does it make sense to go through the cache.

Reply:
@KanishManuja-MS I had initially proposed that the cache be used also inside `GetEdmType()`, but there was some complexity in achieving that, partly because of a recursive call in `GetEdmType` and the method signatures. I don't know if it would be possible to refactor it to use the cache internally without changing public method signatures all over the place (see [this comment](https://github.com/OData/WebApi/pull/2014#issuecomment-573722066), and [this](https://github.com/OData/WebApi/issues/2001#issuecomment-573634346) and [this too](https://github.com/OData/WebApi/issues/2001#issuecomment-573675387)).

However, I could give this another look to see whether we can find a clean way to add the cache in `GetEdmType` without affect too much of the existing methods.

Reply:
@CarlosMontagut
Maybe you could modify `EdmHelpers.GetEdmTypeReference` to call
```
return edmModel.GetTypeMappingCache().GetEdmType(..., ...);
```
and then create a new public method like `EdmHelpers.GetEdmTypeReferenceWithoutCache()` 
method that does exactly what `GetEdmTypeReference` is currently doing. Then `ClrTypeChance.GetEdmType()` should be modified to call `GetEdmTypeReferenceWithoutCache()`. Then the other methods can just call `EdmHelpers.GetEdmTypeReference()` directly and will be using the cache. But `GetEdmModel()` should not be called directly.

@KanishManuja-MS what do you think about that?

Reply:
@habbes, I think we can do that, and hide the use of the Cache inside GetEdmTypeReference, also keeping there a reference to the cache for all the future calls (avoiding the cost of getting the cache each time).

@habbes / @KanishManuja-MS  let me know your thoughts and if you agree I can proceed doing that change.